test/unit/clock_lowres.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2010-2022, Tarantool AUTHORS, please see AUTHORS file.
+ *
+ * Redistribution and use in source and binary forms, with or
+ * without modification, are permitted provided that the following
+ * conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above
+ *    copyright notice, this list of conditions and the
+ *    following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above
+ *    copyright notice, this list of conditions and the following
+ *    disclaimer in the documentation and/or other materials
+ *    provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY <COPYRIGHT HOLDER> ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+ * TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * <COPYRIGHT HOLDER> OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+ * BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ */
+#include "unit.h"
+
+#include "clock_lowres.h"
+#include "clock.h"
+
+static double delta = 0.2;
+
+static inline void
+test_clock(double interval)
+{
+	double lowres_start = clock_lowres();
+	double start = clock_monotonic();
+	while (clock_monotonic() < start + interval) {
+	}
+	double lowres_elapsed = clock_lowres() - lowres_start;
+	ok(lowres_elapsed >= interval - delta &&
+	   lowres_elapsed <= interval + delta,
+	   "Check that lowres clock does not diverge too much from "
+	   "monotonic clock in measuring time interval = %lf", interval);
+}
+
+int
+main(void)
+{
+	clock_lowres_signal_init();
+	plan(4);
+	test_clock(0.3);
+	test_clock(0.7);
+	test_clock(1.0);
+	test_clock(1.3);
```

Comment:
Suggestion: what about a test that runs for say 1 second and checks in a busy loop that the max difference between the value of the low resolution clock and the monotonic clock never exceeds the low resolution clock granularity?

Reply:
Could you please elaborate? Personally, I don't quite understand, what you mean.

Reply:
```C
bool success = true;
while (clock < start + 1) {
    int a = clock_monotonic();
    int b = clock_lowres();
    if (a - b > CLOCK_LOWRES_RESOLUTION) {
        success = false;
        break;
    }
}
```

Reply:
But don't these two clocks have different starting points?

Reply:
Discussed f2f. Turns out we understand the low resolution clock differently. For me, it's a monotonic clock that is updated infrequently (say once every 10 ms). For you it's a completely different counter that doesn't use `clock_monotonic()` at all and instead is incremented every 10 ms. I think using `clock_monotonic()` for updating low resolution clock is better, because:
- We'd have the same concept of monotonic time. We could compare low resolution clock with `ev_now`.
- The low resolution clock wouldn't skew because of accumulated error. Currently, for long intervals, the error may exceed the clock granularity, because it takes some time to process the signal.

Reply:
Sounds good, I will do so.

Reply:
done